dotnet-algorand-sdk/V2/Indexer/Model/Model.cs

```diff
@@ -583,15 +583,15 @@ public partial class BlockUpgradeState
 
         /// <summary>\[nextyes\] Number of blocks which approved the protocol upgrade.</summary>
         [Newtonsoft.Json.JsonProperty("next-protocol-approvals", Required = Newtonsoft.Json.Required.Default, NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
-        public int? NextProtocolApprovals { get; set; }
+        public long? NextProtocolApprovals { get; set; }
```

Comment:
I'm seeing that a lot of these have been changed to long, but I believe that all amounts, rounds, and asset ids should be an unsigned integer with base 64. That would mean that these should really be ulong

Reply:
Agreed , long should be ulong for types derived from things used in TEAL. Let me try to figure out some way of determining what integers should be what and submit a PR for this.

Reply:
i agree